### PR TITLE
[LLVMCPU] Add lowering config for batch_mmt4d

### PR DESCRIPTION
**Pull request by pzread**
Populate lowering config for `batch_mmt4d` op in CPU backend.

Tracking issue: #14431

### CODE REVIEW

**Comment by pzread** [comment]
@RoboTux fyi

File: compiler/src/iree/compiler/Codegen/Interfaces/PartitionableLoopsInterface.cpp [hunk 1]
```diff
@@ -83,6 +83,23 @@ struct Mmt4DOpPartitionableLoops
...
 };
 
+/// External model implementation for linalg::BatchMmt4DOp.
+struct BatchMmt4DOpPartitionableLoops
+    : public PartitionableLoopsInterface::ExternalModel<
+          BatchMmt4DOpPartitionableLoops, linalg::BatchMmt4DOp> {
+  llvm::SmallVector<unsigned>
+  getPartitionableLoops(Operation *op,
+                        std::optional<unsigned> maxNumPartitionedLoops) const {
+    SmallVector<unsigned> parallelLoops = {0, 1, 2};
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Codegen/Interfaces/PartitionableLoopsInterface.cpp:93]: Can we cast the op to linalg::LinalgOp and use `getNumParallelLoops` method?

**Review by hanhanW — APPROVED** [review]

File: compiler/src/iree/compiler/Codegen/Interfaces/PartitionableLoopsInterface.cpp (hunk 1 above)
Comment (pzread) [reply compiler/src/iree/compiler/Codegen/Interfaces/PartitionableLoopsInterface.cpp:93]: I think the issue is `batch_mmt4d` has 5 parallel loops but IIUC we don't want to partition on the last two. That's why the default implementation doesn't work for it (and for `mmt4d`):

https://github.com/openxla/iree/blob/7d7391f1f989d77d1b4c6b02242f51efee93c084/compiler/src/iree/compiler/Codegen/Interfaces/PartitionableLoopsInterface.cpp#L34-L52

Alternatively we can make the default implementation smarter, but I'm not sure about the general criteria of partitionable loops. Might need a little more study

Comment (hanhanW) [reply compiler/src/iree/compiler/Codegen/Interfaces/PartitionableLoopsInterface.cpp:93]: oh, good point. sorry for the wrong suggestion, it looks good to me!

**Pull request merged**